addons/web/static/src/legacy/js/views/calendar/calendar_controller.js

```diff
@@ -445,7 +445,8 @@ var CalendarController = AbstractController.extend({
         if (this.formViewId) {
             options.view_id = parseInt(this.formViewId);
         }
-        new dialogs.FormViewDialog(this, options).open();
+        this.openEventDialog = new dialogs.FormViewDialog(this, options)
```

Comment:
why not create method for prepare options and then override it for timeoff calendar and then pass deletable: true and pass on_remove callback?

Reply:
Thanks @msa-odoo, it looks like a better option indeed :)